### PR TITLE
Update mocha to v3.2.0.

### LINKED ISSUE #1539: Update mocha to latest version
Currently, when doing `npm install` we get various deprecation warnings.

After looking into it:

```
C:\Users\xmr\Desktop\lighthouse>npm list jade
lighthouse@1.4.1 C:\Users\xmr\Desktop\lighthouse
`-- mocha@2.5.3
  `-- jade@0.26.3


C:\Users\xmr\Desktop\lighthouse>npm list to-iso-string
lighthouse@1.4.1 C:\Users\xmr\Desktop\lighthouse
`-- mocha@2.5.3
  `-- to-iso-string@0.0.2


C:\Users\xmr\Desktop\lighthouse>npm list minimatch
lighthouse@1.4.1 C:\Users\xmr\Desktop\lighthouse
+-- babel-core@6.22.1
| `-- minimatch@3.0.3
+-- gulp@3.9.1
| `-- vinyl-fs@0.3.14
|   +-- glob-stream@3.1.18
|   | `-- minimatch@2.0.10
|   `-- glob-watcher@0.0.6
|     `-- gaze@0.5.2
|       `-- globule@0.1.0
|         `-- minimatch@0.2.14
`-- mocha@2.5.3
  `-- glob@3.2.11
    `-- minimatch@0.3.0
```

So, updating mocha should get rid of 3 deprecation warnings.

### PR DESCRIPTION
**Pull request by XhmikosR**
Fixes various `npm i` warnings.

Fixes #1539.

### CODE REVIEW

**Comment by ebidel** [comment]
Google's internal version of mocha is at 2.5.3, so not sure we can use the newer version.

@paulirish @brendankenny I suspect team's don't run mocha though  :)

**Comment by brendankenny** [comment]
Should be fine since it's a dev dependency. The yarn.lock file should be updated in tandem, though, and ideally just the subtree of mocha dependencies, not the full update of #1586...

**Comment by XhmikosR** [comment]
But how can I do that when my local node_modules folder has newer dependencies? Unless you mean manually.

**Comment by brendankenny** [comment]
No, not manually, but it seems like there must be a way...

**Comment by XhmikosR** [comment]
I'll look into it tomorrow.

Is there any reason why you don't like #1586? Because if we add npm-shirnkwrap.json, both would need to be updated regularly.

**Comment by XhmikosR** [comment]
Turns out it was simpler but please verify:

```
C:\Users\xmr\Desktop\lighthouse>yarn add mocha
yarn add v0.19.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 19 new dependencies.
├─ browser-stdout@1.3.0
├─ commander@2.9.0
├─ escape-string-regexp@1.0.5
├─ glob@7.0.5
├─ handlebars@4.0.5
├─ json-stringify-safe@5.0.1
├─ json3@3.3.2
├─ lodash._baseassign@3.2.0
├─ lodash._basecreate@3.0.3
├─ lodash.create@3.1.1
├─ marked@0.3.6
├─ metaviewport-parser@0.0.1
├─ mocha@3.2.0
├─ opn@4.0.2
├─ rimraf@2.2.8
├─ semver@5.3.0
├─ supports-color@3.1.2
├─ whatwg-url@4.0.0
└─ ws@1.1.1
```

**Comment by XhmikosR** [comment]
Now it should be OK. I did `yarn install` in this branch and yarn saved the lock file.

File: yarn.lock [hunk 1]
```diff
@@ -2787,16 +2789,16 @@ webidl-conversions@^3.0.0, webidl-conversions@^3.0.1:
...
+  version "4.0.0"
+  resolved "https://registry.yarnpkg.com/whatwg-url/-/whatwg-url-4.0.0.tgz#5be362f0b6e2f8760f7260df6e0e1df536f5479c"
   dependencies:
     tr46 "~0.0.3"
     webidl-conversions "^3.0.0"
 
-whatwg-url@^4.0.0:
-  version "4.0.0"
-  resolved "https://registry.yarnpkg.com/whatwg-url/-/whatwg-url-4.0.0.tgz#5be362f0b6e2f8760f7260df6e0e1df536f5479c"
+whatwg-url@^3.0.0:
```
Comment (paulirish) [inline yarn.lock]: whatwg-url 3 and 4 changed order here. Not exactly sure why... but probably not a big deal.

**Pull request merged**